### PR TITLE
added __repr__ for annotation classes

### PR DESCRIPTION
**Pull request by grant-Kolena**
### Linked issue(s):
fixes KOL-4420 this is a continuation of this [draft](https://github.com/kolenaIO/kolena/pull/393/files)
### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: kolena/annotation.py [hunk 1]
```diff
@@ -42,15 +43,18 @@ def _data_category() -> str:
         return "ANNOTATION"
 
 
-@dataclass(frozen=True, config=ValidatorConfig)
+@dataclass(frozen=True, config=ValidatorConfig, repr=False)
```
Comment (grant-Kolena) [inline kolena/annotation.py:46]: dataclass creates a default one otherwise

File: kolena/annotation.py [hunk 2]
```diff
@@ -42,15 +43,18 @@ def _data_category() -> str:
...
 class Annotation(TypedDataObject[_AnnotationType], metaclass=ABCMeta):
     """The base class for all annotation types."""
 
     def __init_subclass__(cls, **kwargs):
         _register_data_type(cls)
 
+    def __repr__(self):
```
Comment (consoull) [inline kolena/annotation.py:53]: I suggest using `__str__` instead. That is what pandas is using to dump the object into a csv cell.
It convention that `__repr__` outputs valid python code to construct the object

Comment (consoull) [reply kolena/annotation.py:53]: https://stackoverflow.com/questions/1436703/what-is-the-difference-between-str-and-repr

Comment (consoull) [reply kolena/annotation.py:53]: Nevermind. Containers will use the contained objects __repr__ so we need to use __repr__ here

Comment (grant-Kolena) [reply kolena/annotation.py:53]: Both repr and str needs to be changed after experimentation

**Pull request closed without merging**